NamedPipeWrapper/NamedPipeServer.cs

```diff
@@ -236,13 +276,39 @@ public void Stop()
 				}
 			}
 
+			if (_server != null && _server.IsConnected)
+			{
+				_server.Close();
+			}
+
 			// If background thread is still listening for a client to connect,
 			// initiate a dummy connection that will allow the thread to exit.
 			var dummyClient = new NamedPipeClient<TRead, TWrite>(_pipeName);
 			dummyClient.Start();
 			dummyClient.WaitForConnection(TimeSpan.FromSeconds(2));
 			dummyClient.Stop();
 			dummyClient.WaitForDisconnection(TimeSpan.FromSeconds(2));
+
+		}
+
+		public void Read()
+		{
+			try
+			{
+				if (_server.IsConnected && !_reader.EndOfStream)
+				{
+					string message = _reader.ReadLine();
+					if (message != null)
+					{
+						// Temporary - do something else with the command when we receive it
+						Console.WriteLine(String.Join("", message));
+					}
+				}
+			}
+			catch (Exception e)
+			{
+				Console.WriteLine("Could not read from pipe");
```

Comment:
Are we OK with ignoring the exception and just logging to console? Or should it raise an exception to the hosting app?

Reply:
I see that `private void OnError(Exception exception)` has been utilized to raise read/write exceptions.